src/efi.rs

```diff
@@ -659,4 +670,10 @@ Boot0003* test";
         );
         Ok(())
     }
+    #[test]
+    fn test_get_product_name() -> Result<()> {
+        let name = get_product_name()?;
```

Comment:
I think we shouldn't read global state in a unit test. This will fail on e.g. Debian or other systems and we want to support those.

We can unit test parsing from a `&str` easily enough though.

(That said of course, we could use openat or better cap-std to load from a test root...see what we do in https://github.com/containers/bootc/blob/c866bba1e96e912ab2e36b9eec3985e93e7a0c9b/lib/src/generator.rs#L90 - but unfortunately `OsRelease` doesn't support that yet)

Reply:
> This will fail on e.g. Debian or other systems and we want to support those.

I was wrong, it shouldn't because the `system-release` is optional.

However, it does still make sense to avoid reading global state longer term.

Reply:
Thanks! Yes, you are correct, will switch to `os-release` if not exist `system-release`.